go-app-ussd_public_rapidpro.js

```diff
@@ -182,6 +182,31 @@ go.app = function() {
             return _.intersection(contact_groupids, groups).length > 0;
         };
 
+        self.add = function(name, creator) {
+            self.states.add(name, function(name, opts) {
+                if (self.im.msg.session_event !== 'new')
+                    return creator(name, opts);
+
+                var timeout_opts = opts || {};
+                timeout_opts.name = name;
+                return self.states.create('state_timed_out', timeout_opts);
+            });
+        };
+
+        self.states.add('state_timed_out', function(name, creator_opts) {
+            return new ChoiceState(name, {
```

Comment:
We can probably use a `MenuState` here, instead of having a `next` function that returns `choice.value`